src/dispatch/task/flows.py

```diff
@@ -79,36 +84,46 @@ def send_task_notification(conversation_id, message_template, assignees, descrip
 def create_or_update_task(db_session, incident, task: dict, notify: bool = False):
     """Creates a new task in the database or updates an existing one."""
     # TODO we should standarize this interface (kglisson)
-    creator = task["owner"]
-    assignees = ", ".join(task["assignees"])
+
+    creator = incident_add_or_reactivate_participant_flow(
+        task["owner"], incident_id=incident.id, db_session=db_session
+    )
+
+    assignees = []
+    for a in task["assignees"]:
+        assignees.append(
+            incident_add_or_reactivate_participant_flow(
+                a, incident_id=incident.id, db_session=db_session
+            )
+        )
+
     description = task["description"][0]
     status = TaskStatus.open if not task["status"] else TaskStatus.resolved
     resource_id = task["id"]
     weblink = task["web_link"]
 
     incident_task = task_service.get_by_resource_id(db_session=db_session, resource_id=task["id"])
     if incident_task:
-        if status == TaskStatus.open:
-            # we don't need to take any actions if the status of the task in the collaboration doc is open
+        if incident_task.status == TaskStatus.resolved:
+            # we don't need to take any actions if the task has already been marked as resolved in the database
             return
         else:
-            if incident_task.status == TaskStatus.resolved:
-                # we don't need to take any actions if the task has already been marked as resolved in the database
-                return
-            else:
-                # we mark the task as resolved in the database
-                incident_task.status = TaskStatus.resolved
-                db_session.add(incident_task)
-                db_session.commit()
-
-                if notify:
-                    send_task_notification(
-                        incident.conversation.channel_id,
-                        INCIDENT_TASK_RESOLVED_NOTIFICATION,
-                        assignees,
-                        description,
-                        weblink,
-                    )
+            # we mark the task as resolved in the database
+            incident_task.status = TaskStatus.resolved
+
+            if notify:
+                send_task_notification(
+                    incident.conversation.channel_id,
+                    INCIDENT_TASK_RESOLVED_NOTIFICATION,
+                    assignees,
+                    description,
+                    weblink,
+                )
+
+        # always sync assignees
+        incident_task.assignees = assignees
+        db_session.add(incident_task)
+        db_session.commit()
```

Comment:
I think the logic we had before was correct. You need to check the status of the task that gets passed, not the one in the database. If the status is open, then we don't need to update it and we return. If already resolved in the doc and database, then we return. If not resolved in the db, then we resolve it.